src/main/clm_driver.F90

```diff
@@ -1045,6 +1028,41 @@ subroutine clm_drv(doalb, nextsw_cday, declinp1, declin, rstwr, nlend, rdate, ro
                 c14_soilbiogeochem_carbonflux_inst, c14_soilbiogeochem_carbonstate_inst, &
                 soilbiogeochem_nitrogenflux_inst, soilbiogeochem_nitrogenstate_inst,     &
                 clm_fates, nc)
+
+          call clm_fates%wrap_update_hifrq_hist(bounds_clump, &
+               soilbiogeochem_carbonflux_inst, &
+               soilbiogeochem_carbonstate_inst)
+
+          
+          if( is_beg_curr_day() ) then
+
+             ! --------------------------------------------------------------------------
+             ! This is the main call to FATES dynamics
+             ! --------------------------------------------------------------------------
+
+             if ( masterproc ) then
+                write(iulog,*)  'clm: calling FATES model ', get_nstep()
+             end if
+             
+             call clm_fates%dynamics_driv( nc, bounds_clump,                        &
```

Comment:
Is it due to the addition of `UnPackNutrientAquisitionBCs` into `dynamics_driv`?

Reply:
Two reasons.  At some point when we fully couple nutrients with CTSM-FATES, we will ask the CTSM native nutrient and decomposition code to also accommodate FATES, intead of having our own call.  The native call sequence actually has all the stuff happen before the dynamics step, which this prepares us for.  Also, because of this first point, and because of a call to the high-frequency history stuff, there is an order dependance on when this stuff happens to correctly write the history.

Reply:
Thanks for the explanation.